### PR TITLE
Ensure compatibility with PHP 7.4 arrow functions

### PR DESCRIPTION
**Pull request by julienfalque**
Related to #4382.

### CODE REVIEW

**Comment by julienfalque** [comment]
Test added, thank you @gharlan.

**Comment by keradus** [comment]
(i did rebase + squash + fix minor issues)

File: tests/Fixer/FunctionNotation/MethodArgumentSpaceFixerTest.php [hunk 1]
```diff
@@ -1021,6 +1021,39 @@ functionCall(
...
+
+    public function provideFix74Cases()
+    {
+        return [
+            [
+                '<?php
+$fn = fn(
+    $test1,
+    $test2
+) => null;',
```
Comment (keradus) [inline tests/Fixer/FunctionNotation/MethodArgumentSpaceFixerTest.php]: unfortunately this case is failing. can you work on it, @julienfalque ?

Comment (julienfalque) [reply tests/Fixer/FunctionNotation/MethodArgumentSpaceFixerTest.php]: Fixed.

Comment (keradus) [reply tests/Fixer/FunctionNotation/MethodArgumentSpaceFixerTest.php]: big thanks!

**Comment by keradus** [comment]
Thank you @julienfalque.

**Comment by TobiKlaas** [comment]
Unfortunately, this is not working. With --verbose -> invalid file syntax.

> return fn(SearchHit $hit) => $hit->valueObject;

Not accepted, quite frustrating!

**Pull request merged**